File: C++/섹션04. 자료구조/04. BOJ 2161 - 카드1.cpp
#include <stdio.h>

#define MAX (1000 + 100)

int N;
int queue[MAX * 2];
int rp, wp;

int main()
{	
	scanf("%d", &N);

	rp = wp = 0;

	for (int i = 1; i <= N; i++) queue[wp++] = i;

	for (int i = 0; i < N - 1; i++)
	{
		printf("%d ", queue[rp++]);

		queue[wp++] = queue[rp++];
	}

	printf("%d\n", queue[wp - 1]); // rp
	
	// printf("=======================\n");

	// for (int i = 0; i < wp; i++) printf("%d ", queue[i]);
	// putchar('\n');

	return 0;
}

File: Python/섹션04. 자료구조/04. BOJ 2161 - 카드1.py
MAX = 1000 + 100

queue = [0] * MAX * 2
rp, wp = 0, 0

N = int(input())

for i in range(1, N + 1):
    queue[wp] = i
    wp += 1
    
for _ in range(N - 1):
    print(queue[rp], end=' ')
    rp += 1
    
    queue[wp] = queue[rp]
    rp += 1
    wp += 1
    
print(queue[wp - 1])